detect/detect.go

```diff
@@ -201,8 +201,12 @@ func (d *Detector) detectRule(fragment Fragment, rule *config.Rule) []report.Fin
 		}
 
 		// check if the secret is in the allowlist
-		if rule.Allowlist.RegexAllowed(finding.Secret) ||
-			d.Config.Allowlist.RegexAllowed(finding.Secret) {
+		regexMatchText := finding.Secret
+		if d.Config.Allowlist.RegexFullLine {
+			regexMatchText = fragment.Raw
```

Comment:
@zanven42 `fragment.Raw` could be multiline, you would have to use `fragment.newLineIndices` and some of the code in the `location` helper function to determine the line content

Reply:
good point, I had a think regarding this and wonder if it is more ideal for future proofing to have this feature check against the entire match including multi line cases and update the documentation and wording of variables to align.

Because i imagine one day someone will complain they can't get multi line checks to work and then either this feature is deprecated or it gets more complex with 2 booleans to turn on.

Reply:
it could be my understanding of "fragment.Raw" is wrong and it returns things outside of the scope of the match.